esti/s3_gateway_test.go

```diff
@@ -289,6 +289,88 @@ func TestS3IfNoneMatch(t *testing.T) {
 	}
 }
 
+func TestListMultipartUploads(t *testing.T) {
+	ctx, logger, repo := setupTest(t)
+	defer tearDownTest(repo)
+	s3Endpoint := viper.GetString("s3_endpoint")
+	s3Client := createS3Client(s3Endpoint, t)
+	multipartNumberOfParts := 3
+	multipartPartSize := 5 * 1024 * 1024
+
+	blockStoreType := viper.GetString(ViperBlockstoreType)
+	if blockStoreType != "s3" {
+		t.Skip("Skipping test - blockstore type is not s3")
+	}
+
+	// create two objects for two mpus
+	obj1 := "object1"
+	obj2 := "object2"
+	path1 := "main/" + obj1
+	path2 := "main/" + obj2
+
+	input1 := &s3.CreateMultipartUploadInput{
+		Bucket: aws.String(repo),
+		Key:    aws.String(path1),
+	}
+	input2 := &s3.CreateMultipartUploadInput{
+		Bucket: aws.String(repo),
+		Key:    aws.String(path2),
+	}
+	// create first mpu
+	resp1, err := s3Client.CreateMultipartUpload(ctx, input1)
+	require.NoError(t, err, "failed to create multipart upload")
+	parts := make([][]byte, multipartNumberOfParts)
+	for i := 0; i < multipartNumberOfParts; i++ {
+		parts[i] = randstr.Bytes(multipartPartSize + i)
+	}
+
+	completedParts1 := uploadMultipartParts(t, ctx, s3Client, logger, resp1, parts, 0)
+
+	completeInput1 := &s3.CompleteMultipartUploadInput{
+		Bucket:   resp1.Bucket,
+		Key:      resp1.Key,
+		UploadId: resp1.UploadId,
+		MultipartUpload: &types.CompletedMultipartUpload{
+			Parts: completedParts1,
+		},
+	}
+	// check first mpu appears
+	output, err := s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	require.NoError(t, err, "error listing multiparts")
+	str := concatKeys(output)
+	require.Contains(t, str, obj1)
+
+	// create second mpu check both appear
+	_, err = s3Client.CreateMultipartUpload(ctx, input2)
+	require.NoError(t, err, "failed to create multipart upload")
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	str = concatKeys(output)
+	fmt.Println(str)
+	require.Contains(t, str, obj1)
+	require.Contains(t, str, obj2)
+
+	// finish first mpu check only second appear
+	_, err = s3Client.CompleteMultipartUpload(ctx, completeInput1)
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	require.NoError(t, err, "error listing multiparts")
+	str = concatKeys(output)
+	require.NotContains(t, str, obj1)
+	require.Contains(t, str, obj2)
+}
+
+func concatKeys(output *s3.ListMultipartUploadsOutput) string {
+	if output == nil {
+		return ""
+	}
+	var allKeys string
+	for _, upload := range output.Uploads {
+		if upload.Key != nil {
+			allKeys += *upload.Key + " "
+		}
+	}
+	return allKeys + "\n"
+}
```

Comment:
Just passing through, not a full review.

It is not a good option to use the full string of the results (keys) and assert substring search as values can be sub values of others.

Return a []string with the values (keys) you like to extract
Use the same assert func Contains should work the same - but this time it will exact result and more verbose message in case the assert fail.
Better func name can be `extractUploadKeys`

```go
func extractUploadKeys(output *s3.ListMultipartUploadsOutput) []string {
	if output == nil {
		return nil
	}
	keys : = make([]string, 0, len(output.Uploads))
	for _, upload := range output.Uploads {
		if upload.Key != nil {
			keys = append(keys, *upload.Key)
		}
	}
	return keys
}
```